### PR TITLE
WIFI-2381: Query DigiCert's API for Redirector's address

### PR DESCRIPTION
**Pull request by brennerm**
Signed-off-by: Max Brenner <xamrennerb@gmail.com>

### CODE REVIEW

**Review by leonidlm — APPROVED** [review]

File: feeds/wlan-ap/opensync/files/bin/wlan_ap_redirector.sh
Comment (RickSommerville) [inline feeds/wlan-ap/opensync/files/bin/wlan_ap_redirector.sh:5]: The DigiCert option depends on the existence of  AP_DEVICE_ID_FILE="/usr/opensync/certs/client_deviceid.txt".
How does this get created?  It does not exist on my CIG WF194c.
Should the code check for the existence of the file and error out instead of doing a call to DigiCert that will fail?

Comment (brennerm) [reply feeds/wlan-ap/opensync/files/bin/wlan_ap_redirector.sh:5]: This file is part of the AP package that is being created by the PKI scripts as you can see [here](https://github.com/Telecominfraproject/wlan-pki-cert-scripts/blob/digicert/digicert-generate-ap-certs.sh#L72).

Will add a file existence check. :+1:

Comment (RickSommerville) [reply feeds/wlan-ap/opensync/files/bin/wlan_ap_redirector.sh:5]: Thanks for the link.  Should this other branch also have a PR or be part of this PR?  Once you add the file check, do you want this PR merged alone?

File: feeds/wlan-ap/opensync/files/bin/wlan_ap_redirector.sh
Comment (brennerm) [inline feeds/wlan-ap/opensync/files/bin/wlan_ap_redirector.sh:27]: @RickSommerville Do you know if `jq` is available on a stock firmware image?

File: feeds/wlan-ap/opensync/files/bin/wlan_ap_redirector.sh
Comment (brennerm) [reply feeds/wlan-ap/opensync/files/bin/wlan_ap_redirector.sh:5]: "generate data to be loaded to the AP" <- this

We need to wait for the merge in the PKI scripts repo before we can merge this PR.

File: feeds/wlan-ap/opensync/files/bin/wlan_ap_redirector.sh
Comment (RickSommerville) [reply feeds/wlan-ap/opensync/files/bin/wlan_ap_redirector.sh:27]: I don't think so.  I tried to just execute 'jq' from SSH to AP and it reports "-ash: jq: not found".
"which jq" similarly reports nothing.

Comment (brennerm) [reply feeds/wlan-ap/opensync/files/bin/wlan_ap_redirector.sh:27]: Mmh do you think it's feasible to include it in the image? Otherwise we will have trouble to reliably extract the redirector URL from the Digicert response.

Comment (brennerm) [reply feeds/wlan-ap/opensync/files/bin/wlan_ap_redirector.sh:27]: Or maybe you know of any other tool that is already installed and capable of handling JSON.

Comment (RickSommerville) [reply feeds/wlan-ap/opensync/files/bin/wlan_ap_redirector.sh:27]: Look at wan-ap-config/etc/uci-defaults/90-tip-hostname.  It uses "jsonfilter".  Relevant code is:

MAC=$(cat /etc/board.json | jsonfilter -e '@["network"]["wan"]["macaddr"]' | tr -d :)

Comment (brennerm) [reply feeds/wlan-ap/opensync/files/bin/wlan_ap_redirector.sh]: Awesome, thanks! Replaced the query using jsonfilter.

**Comment by brennerm** [comment]
@RickSommerville Will this script get executed automatically on boot?

**Comment by RickSommerville** [comment]
Merged to pending a1ff0f90

**Pull request closed without merging**
